atomicapp/providers/kubernetes.py

```diff
@@ -38,227 +44,243 @@ class KubernetesProvider(Provider):
+            self._process_artifact_data(artifact, data)
 
-    def _call(self, cmd):
-        """Calls given command
+    def _process_artifact_data(self, artifact, data):
+        """
+        Process the data for an artifact
 
-        :arg cmd: Command to be called in a form of list
-        :raises: Exception
+        Args:
+            artifact (str): Artifact name
+            data (dict): Artifact data
         """
 
-        if self.dryrun:
-            logger.info("DRY-RUN: %s", " ".join(cmd))
-        else:
-            ec, stdout, stderr = Utils.run_cmd(cmd, checkexitcode=True)
-            return stdout
+        # Check if kind exists
+        if "kind" not in data.keys():
+            raise ProviderFailedException(
+                "Error processing %s artifact. There is no kind" % artifact)
 
-    def process_k8s_artifacts(self):
-        """Processes Kubernetes manifests files and checks if manifest under
-        process is valid.
-        """
-        for artifact in self.artifacts:
-            data = None
-            with open(os.path.join(self.path, artifact), "r") as fp:
-                logger.debug(os.path.join(self.path, artifact))
-                try:
-                    data = anymarkup.parse(fp)
-                except Exception:
-                    msg = "Error processing %s artifcats, Error:" % os.path.join(
-                        self.path, artifact)
-                    cockpit_logger.error(msg)
-                    raise
-            if "kind" in data:
-                self.k8s_manifests.append((data["kind"].lower(), artifact))
-            else:
-                apath = os.path.join(self.path, artifact)
-                raise ProviderFailedException("Malformed kube file: %s" % apath)
+        # Change to lower case so it's easier to parse
+        kind = data["kind"].lower()
 
-    def _resource_identity(self, path):
-        """Finds the Kubernetes resource name / identity from resource manifest
-        and raises if manifest is not supported.
+        if kind not in self.k8s_artifacts.keys():
+            self.k8s_artifacts[kind] = []
 
-        :arg path: Absolute path to Kubernetes resource manifest
+        # Fail if there is no metadata
+        if 'metadata' not in data:
+            raise ProviderFailedException(
+                "Error processing %s artifact. There is no metadata object" % artifact)
 
-        :return: str -- Resource name / identity
+        # Change to the namespace specified on init()
+        data['metadata']['namespace'] = self.namespace
 
-        :raises: ProviderFailedException
+        if 'labels' not in data['metadata']:
+            data['metadata']['labels'] = {'namespace': self.namespace}
+        else:
+            data['metadata']['labels']['namespace'] = self.namespace
+
+        self.k8s_artifacts[kind].append(data)
+
+    '''
+    This is DEPRECATED and not needed anymore as we check the /resource URL of the kubernetes api against the artifact
```

Comment:
we'll eventually remove this - but thanks for leaving it in.. 

can you elaborate? what code are we doing now that we weren't in the past?  i.e. where do we "check the /resource URL" and why does doing that make this no longer needed. 

@tkral may be interested in this